Add Duration to Defense Debuff Messages

samisbakedham opened the pull request:
## Description:

This PR updates the in-game defense debuff message to include the **duration** of the effect, improving clarity and usability during gameplay.

### Change Summary:
- Updated static text to:  
  `You broke your alliance with United States, making you a TRAITOR (50% defense debuff for 30 secs)`
- Improves player understanding of how long penalties last.
- Aligns with other UI efforts focused on real-time feedback and clearer mechanics.
- Sets a standard for future status/debuff messages to include duration info where relevant.

This small but high-impact change enhances the player experience, especially in intense or competitive matches where timing matters.

---

## Please complete the following:

- [] I have added screenshots for all UI updates  
- [x] I confirm I have thoroughly tested these changes and take full responsibility for any bugs introduced  
- [x] I understand that submitting code with bugs that could have been caught through manual testing blocks releases and new features for all contributors  

---

## Please put your Discord username so you can be contacted if a bug or regression is found:

**@walker9607**


Resolves issue https://github.com/openfrontio/OpenFrontIO/issues/602#issue-3018752308

joibor commented:
Is it a good suggestion.

evanpelle commented on src/client/graphics/layers/EventsDisplay.ts:
```diff
@@ -274,7 +274,7 @@ export class EventsDisplay extends LitElement implements Layer {
         description:
           `You broke your alliance with ${betrayed.name()}, making you a TRAITOR ` +
-          `(${malusPercent}% defense debuff)`,
+          `(${malusPercent}% defense debuff for 30 seconds)`,
```
can you do for ${this.game.config().traitorDuation() / 10}

that way this updates if we change how long the traitor duration is.

samisbakedham replied:
Yes

samisbakedham commented:
@evanpelle @APuddle210 
Updated the betrayal event message to dynamically show the current traitor defense debuff and duration based on this.game.config().traitorDuration.

Implemented singular/plural grammar handling (e.g., "1 second" vs. "30 seconds").

Preserved and improved the original betrayal message ("You broke your alliance with ${betrayed.name()}, making you a TRAITOR") by appending debuff information.

Fixed syntax issues and removed redundant/invalid code from onBrokeAllianceEvent.

Ensured future updates to traitor debuff duration are automatically reflected without manual message edits.

evanpelle commented:
please run `npm run format`

samisbakedham commented:
> npm run format

Done

evanpelle reviewed (approved):
Thanks!
